pro/src/components/layout/CsvTable/CsvTable.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FunctionComponent, useEffect, useState } from 'react'
+import { useLocation } from 'react-router-dom'
+
+import { API_URL } from 'utils/config'
+import { getKey } from 'utils/keys'
+
+import Spinner from '../Spinner'
+
+import { getCsvData } from './csvService'
+import type { CsvData } from './csvService'
+
+const CsvTable: FunctionComponent = () => {
+  const { search } = useLocation()
+
+  const handlePrintCurrentView = () => window.print()
+  const [dataFromCsv, setDataFromCsv] = useState<CsvData | null | undefined>()
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(search)
+    getCsvData(
+      `${API_URL}/reimbursements/csv?venueId=${searchParams.get(
+        'venueId'
+      )}&reimbursementPeriodBeginningDate=${searchParams.get(
+        'reimbursementPeriodBeginningDate'
+      )}&reimbursementPeriodEndingDate=${searchParams.get(
+        'reimbursementPeriodEndingDate'
+      )}`
+    )
+      .then(res => {
+        setDataFromCsv(res)
+      })
+      .catch(() => {
+        setDataFromCsv(null)
+      })
+      .finally(() => setIsLoading(false))
+  }, [search])
+
+  return (
+    <>
+      {isLoading && (
+        <div id="spinner-container">
+          <Spinner />
+        </div>
+      )}
+
+      {!isLoading && dataFromCsv?.data?.length && (
```

Comment:
```suggestion
  if (isLoading) {
    return (
      <div id="spinner-container">
          <Spinner />
       </div>
    )
  }
  return (
    <>
      {dataFromCsv?.data?.length && (
      // [...]
      {(!dataFromCsv?.data || dataFromCsv?.data?.length === 0) && (
```

On devrait aussi pouvoir simplifier les deux autres conditions: 

```javascript
      const hasResults = dataFromCsv?.data?.length > 0
      // ...
      {hasResults ? (
        // display results table
      ) : (
        // display no results message
      )
```